.github/workflows/cabal-ci.yml

```diff
@@ -0,0 +1,20 @@
+on:
+  - push
+  - pull_request
+name: cabal build
+jobs:
+  runhaskell:
+    name: cabal test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+      - uses: haskell/actions/setup@v2
+        with:
+          ghc-version: '9.4.1'
+          cabal-version: '3.8.1.0'
+      - run: cat cabal.project | grep -E '^packages|lens$|runtime$' > new.cabal.project
+      - run: mv new.cabal.project cabal.project
+      - run: cabal build all
+      - run: cabal test proto-lens
```

Comment:
An alternative here is to have a `ghc94.cabal.project` and use `cabal build --project-file ghc94.cabal.project all`. That way this step can be checked in and maintained separately.

Reply:
Oh, great! I have applied your suggestion.